client-v2/src/main/java/com/clickhouse/client/api/data_formats/internal/SerializerUtils.java

```
@@ -462,6 +462,12 @@ private static void serializeMapData(OutputStream stream, Object value, ClickHou
     }
 
     private static void serializePrimitiveData(OutputStream stream, Object value, ClickHouseColumn column) throws IOException {
+        //Handle null values
+        if (value == null && column.isNullable()) {//Only nullable columns can have null values
```

If it null and not nullable should we throw an exception like data is not valid